treesit-auto.el

```diff
@@ -112,11 +121,71 @@ remap the tree-sitter variant back to the default mode."
            (when name-mode-bound-p
              (add-to-list 'major-mode-remap-alist `(,name-ts-mode . ,name-mode)))))))
 
+(defun treesit-auto--prompt-to-install-package (lang &optional ignore-prompt)
+  "Ask the user if they want to install a treesitter grammar for `LANG'.
+
+If `IGNORE-PROMPT' is set to true, no prompt will be given and we
+will try to install the tree-sitter grammar as long as a
+repository can be found in `treesit-language-source-alist'.
+
+Returns `non-nil' if install was completed without error."
+  (if-let* ((repo (alist-get lang treesit-language-source-alist))
+            (response (or ignore-prompt
+                          (yes-or-no-p (format "Tree Sitter grammar for %s is missing.  Would you like to install it from: %s?"
+                                               (symbol-name lang)
+                                               (car repo))))))
+      ;; treesit-install-language-grammar will return nil if the
+      ;; operation succeeded and 't if a warning was tossed. I don't
+      ;; think this is by design but just because of the way
+      ;; `display-warning' works.
+      (progn
+        (message "Installing the tree-sitter grammar for %s" lang)
+        (not (treesit-install-language-grammar lang)))))
```

Comment:
Suggestion:  I wonder if we can simplify the signature to this function by using `cond` against the value of `treesit-auto-install`?  Something like this:

```suggestion
(defun treesit-auto--prompt-to-install-package (lang)
  "Ask the user if they want to install a treesitter grammar for `LANG'.
If `IGNORE-PROMPT' is set to true, no prompt will be given and we
will try to install the tree-sitter grammar as long as a
repository can be found in `treesit-language-source-alist'.
Returns `non-nil' if install was completed without error."

  (let ((repo (alist-get lang treesit-language-source-alist)))
    (when (cond ((eq t treesit-auto-install) t)
                ((eq 'prompt treesit-auto-install)
                 (yes-or-no-p (format "Tree Sitter grammar for %s is missing.  Would you like to install it from: %s?"
                                      (symbol-name lang)
                                      (car repo))))
                (t) nil)
      (message "Installing the tree-sitter grammar for %s" lang)
      ;; treesit-install-language-grammar will return nil if the
      ;; operation succeeded and 't if a warning was tossed. I don't
      ;; think this is by design but just because of the way
      ;; `display-warning' works.
      (not (treesit-install-language-grammar lang)))))
```

Then, below, where we call `treesit-auto--prompt-to-install-package`, we can omit the `(not (equal ...` argument altogether.

Reply:
Yeah I'm totally cool, doing it this way - I just default to passing data into the function.